tests/conftest.py

```diff
@@ -26,3 +27,6 @@ def cleanup_children():
         for child in children:
             with suppress(psutil.NoSuchProcess):
                 child.kill()
+
+    # Killing child processes may leave the MPFuture backend broken
+    MPFuture.reset_backend()
```

Comment:
Perhaps it's only necessary to do this when we had to terminate some child processes by force?
Solution 2: it might be preferable to reser the global state between the tests regardless of whether we actually kill processes. In that case, an explicit cleanup might still be useful, but a comment can be more general

Reply:
I weakly support option 2, rationale: if some fail in such a way that backend is corrupted, resetting backend ensures that no other tests will be affected, hence we will know the main culprit right away

Reply:
I've implemented the 2nd option (updated the comment).